### PR TITLE
[PM-9552] add a toggle for the routing animation

### PR DESCRIPTION
**Pull request by slonkazoid**
## 🎟️ Tracking

https://github.com/orgs/bitwarden/discussions/9995

## 📔 Objective

there is currently a really annoying 400ms animation whenever you click on anything in the extension. besides being annoying it is an accessibility issue  
there is already an [old and exceedingly popular community thread](https://community.bitwarden.com/t/option-to-adjust-or-disable-ui-animation/10614) about this

## 📸 Screenshots

image

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

## TODO

i am not experienced at Angular, so i can not give any guarantees about the correctness or quality of this code. i only took action because no one else did

- [ ] Internationalization
- ..?

### CODE REVIEW

**Review by coroiu — COMMENTED** [review]
Some smaller comments. The PR also has some linting issues that need to be addressed :)

File: libs/common/src/platform/animations/animation-control.service.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+   * The routing animation toggle.
+   */
+  abstract routingAnimation$: Observable<boolean>;
+
+  /**
+   * A method for updating the state of the animation toggle.
+   * @param theme The new state.
+   */
+  abstract setRoutingAnimation(state: boolean): Promise<void>;
+}
```
[the remark this answers is not in the document]
Comment (slonkazoid) [reply libs/common/src/platform/animations/animation-control.service.ts]: also done

File: apps/browser/src/_locales/en/messages.json [hunk 2]
```diff
@@ -3570,5 +3570,10 @@
         "example": "Visa"
       }
     }
+  },
+  "enableRoutingAnimation": {
+    "message": "Enable routing animations"
+  },"enableRoutingAnimationDesc": {
+    "message": "Enable the sliding animation when you click on entries and such."
```
Comment (danielleflinn) [inline apps/browser/src/_locales/en/messages.json:3577]: Let's remove this text as the checkbox label is clear enough. (We have some work coming up that will simplify a lot of the text you see on this page).

File: apps/browser/src/_locales/en/messages.json [hunk 3]
```diff
@@ -3570,5 +3570,10 @@
         "example": "Visa"
       }
     }
+  },
+  "enableRoutingAnimation": {
+    "message": "Enable routing animations"
```
[the remark this answers is not in the document]
Comment (slonkazoid) [reply apps/browser/src/_locales/en/messages.json:3575]: should i rename the variables as well?

File: apps/browser/src/_locales/en/messages.json (hunk 2 above)
Comment (slonkazoid) [reply apps/browser/src/_locales/en/messages.json]: done

File: apps/browser/src/_locales/en/messages.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (danielleflinn) [reply apps/browser/src/_locales/en/messages.json]: I think renaming the messages.json variable makes sense to match, but I'll leave that up to @coroiu to make the final call

[the remark this answers is not in the document]
Comment (slonkazoid) [reply apps/browser/src/_locales/en/messages.json]: also, the current behavior doesn't match the description. will it apply to the autofill animation now, or in the future?

[the remark this answers is not in the document]
Comment (danielleflinn) [reply apps/browser/src/_locales/en/messages.json]: We'll likely explore this in the future–we also may add other types of animations in the future that I'd like it to be easy enough for engineers to leverage this toggle to disable when they first implement.

**Comment by nick-livefront** [comment]
Hi, @slonkazoid! 

Could you pull in the latest from `main` and fix the conflicts?

Just as an FYI, there is a UI refresh of the browser extension in the works and a new appearance setting page that I merged [#10458](https://github.com/bitwarden/clients/pull/10458). These changes are hidden behind the `ExtensionRefresh` feature flag.

I'm working to determine if we should apply your changes to the existing _and_ the refresh extension or just the extension refresh and it will go out when that rolls out. I can follow up here when I know more!

cc: @danielleflinn

**Comment by slonkazoid** [comment]
should be it, but i havent test-built with the changes. i will do that tomorrow

**Comment by slonkazoid** [comment]
that seems to have done it

**Review by shane-melton — APPROVED** [review]
Thanks for the contribution @slonkazoid!  This has been approved and is being handed to QA for testing before we merge.

**Review by justindbaur — APPROVED** [review]

**Pull request merged**